### PR TITLE
EPMEDU-731 - Updating requesting GPS permission and GPS location

### PR DESCRIPTION
**Pull request by Andrii-Chornyi**
PR handles EPMEDU-731 only partially. New PR in scope of same ticket for handling preselection of feeding points based on remoteness is required.

Updating logic around accessing GPS permissions and enabling GPS locations.
- Added pink background for selected feeding point.

https://user-images.githubusercontent.com/112079677/234331590-eec5f938-109a-4b43-90e7-8f02779886b7.mp4

- Updated GPS location request to use inbuilt android dialog instead of going to phone location settings.
- Fixed bug when user wasn't able to center himself on the map by clicking "Find me" button after providing GPS permission on first launch.
- Fixed multiple "fetchGpsSettingsUpdates().collect(::collectGpsSettings)" coroutines launches by setting LaunchedEffect with proper keys at GeoLocationPermission

https://user-images.githubusercontent.com/112079677/231162265-43fafbe2-1cb2-4c7c-b780-d1287a9b2900.mp4

### CODE REVIEW

File: library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt [hunk 1]
```diff
@@ -39,6 +46,36 @@ fun Context.launchGpsSettings() {
...
+    LocationServices
+        .getSettingsClient(this)
+        .checkLocationSettings(builder.build())
+        .addOnFailureListener(getActivity()!!) { resolvableException ->
+
+            if (resolvableException is ResolvableApiException) {
+                val dummyRequestCode = 9999
+                // Show the "Enable Location" dialog by calling startResolutionForResult()
+                resolvableException.startResolutionForResult(
+                    getActivity()!!,
```
Comment (VEZE) [inline library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt]: remove !! expression

File: library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt [hunk 2]
```diff
@@ -39,6 +46,36 @@ fun Context.launchGpsSettings() {
...
+
+    val builder = LocationSettingsRequest.Builder()
+        .addLocationRequest(locationRequest)
+
+    /** Locations service is expected to fail. Inside of Exception of onFailureMethod
+     * Service which provides logic for calling "Enable Location" dialog is returned */
+    LocationServices
+        .getSettingsClient(this)
+        .checkLocationSettings(builder.build())
+        .addOnFailureListener(getActivity()!!) { resolvableException ->
```
Comment (VEZE) [inline library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt]: remove !! expression

Comment (VEZE) [reply library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt]: addOnFailureListener can be used without getActivity()

File: library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt [hunk 3]
```diff
@@ -39,6 +46,36 @@ fun Context.launchGpsSettings() {
...
+
+            if (resolvableException is ResolvableApiException) {
+                val dummyRequestCode = 9999
```
Comment (VEZE) [inline library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt]: move to const val, use appropriate name

File: library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt [hunk 4]
```diff
@@ -39,6 +46,36 @@ fun Context.launchGpsSettings() {
     startActivity(intent)
 }
 
+/** Workaround method to request GPS location using dialog provided by Google */
+fun Context.requestGpsByDialog() {
+    val dummyInterval = 1000L
```
Comment (VEZE) [inline library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt]: move to const val, use appropriate name

File: library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt (hunk 2 above)
Comment (VEZE) [inline library/extensions/src/main/java/com/epmedu/animeal/extensions/Context.kt]: naming -> resolvableException

**Review by VEZE — CHANGES_REQUESTED** [review]

**Comment by Fileek** [comment]
It turned out that we have to implement auto-select of the closest feeding point. 
But let's do it after implementing taking and attaching photos.

**Review by Fileek — APPROVED** [review]

**Pull request merged**